### PR TITLE
#1579 - vDisk cache management

### PR DESCRIPTION
**Pull request by kvanhijf**

### CODE REVIEW

File: ovs/lib/monitoring.py [hunk 1]
```diff
@@ -55,3 +60,64 @@ def test_ssh_connectivity():
...
+            if alba_backend_guid not in alba_guid_size_map:
+                ovs_client = OVSClient(ip=alba_backend_host,
+                                       port=connection_info['port'],
+                                       credentials=(connection_info['client_id'], connection_info['client_secret']),
+                                       version=2)
+                try:
+                    alba_guid_size_map[alba_backend_guid] = {'name': alba_backend_name,
+                                                             'used_size': 0,
+                                                             'backend_ip': alba_backend_host,
+                                                             'total_size': ovs_client.get('/alba/backends/{0}/'.format(alba_backend_guid), params={'contents': 'usages'})['usages']['free']}
```
Comment (khenderick) [inline ovs/lib/monitoring.py]: Better rename this to `free`.

Comment (khenderick) [reply ovs/lib/monitoring.py]: Better just use the actual `total`, it doesn't make sense to use `free`.

File: ovs/lib/monitoring.py [hunk 2]
```diff
@@ -55,3 +60,64 @@ def test_ssh_connectivity():
...
+            backend_ip = alba_backend_info['backend_ip']
+
+            location = 'local'
+            remote_msg = ''
+            if backend_ip not in local_ips:
+                location = 'remote'
+                remote_msg = ' (on remote IP {0})'.format(backend_ip)
+            percentage = used / total * 100
+            if percentage > 100:
+                MonitoringController._logger.error('OVS_ERROR: Over-allocation for vDisk caching quota on {0} ALBA Backend {1}{2}. Unexpected behavior might occur'.format(location, name, remote_msg))
```
Comment (khenderick) [inline ovs/lib/monitoring.py]: Let's make this a `OVS_WARNING` for now.

File: webapps/frontend/app/viewmodels/wizards/addvdisk/gather.js [hunk 3]
```diff
@@ -229,5 +218,74 @@ define([
...
+                        if (self.loading() === true) {
+                            self.loading(false);
+                        }
+                    });
+            }, 5000);
+            self.refresher.run();
+            self.refresher.start();
+        };
+        // self.deactivate = function() {
+        //     self.vPoolSubscription.dispose();
```
Comment (khenderick) [inline webapps/frontend/app/viewmodels/wizards/addvdisk/gather.js]: Should be finished

File: webapps/frontend/app/viewmodels/wizards/addvdisk/gather.js [hunk 4]
```diff
@@ -229,5 +218,74 @@ define([
...
         };
+
+        // Subscriptions
+        self.nameSubscription = self.data.name.subscribe(function() {
+            if (!self.canContinue().value) {
+                self.preValidateResult({reasons: [], fields: []});  // Reset the pre-validate result in case a vDisk name was specified which already existed
+            }
+        });
+        self.vPoolSubscription = self.data.vPool.subscribe(function (vPool) {
+            if (vPool === undefined || vPool.metadata() === undefined || !vPool.metadata().hasOwnProperty('backend')) {
```
Comment (khenderick) [inline webapps/frontend/app/viewmodels/wizards/addvdisk/gather.js]: Does it work correctly if a user continuously switches the selected vPool?

File: webapps/frontend/app/views/site/vpool-detail.html [hunk 5]
```diff
@@ -157,10 +157,9 @@
                         </tr>
                         <tr>  <!-- row 6 -->
                             <td data-i18n="ovs:generic.nrofvdisks"></td>
-                            <td><span data-bind="text: vDisks().length"></span></td>
+                            <td data-bind="text: vDisks().length"></td>
                             <td data-i18n="ovs:generic.write_speed"></td>
-                            <td><span data-bind="text: backendWriteSpeed"></span></td>
-                            <td colspan="2" >&nbsp;</td>
```
Comment (khenderick) [inline webapps/frontend/app/views/site/vpool-detail.html]: Restore this one.

**Review by khenderick — CHANGES_REQUESTED** [review]

**Review by khenderick — APPROVED** [review]

**Comment by kvanhijf** [comment]
openvstorage-2.8.2-dev.1492591068.7d6addd

**Pull request merged**